Base/QTGUI/qSlicerApplication.cxx

```diff
@@ -141,8 +141,8 @@ class qSlicerApplicationPrivate : public qSlicerCoreApplicationPrivate
   QSettings* newSettings() override;
 
   QPointer<qSlicerLayoutManager> LayoutManager;
-  ctkToolTipTrapper*      ToolTipTrapper;
-  ctkSettingsDialog*      SettingsDialog;
+  ctkToolTipTrapper* ToolTipTrapper;
+  QPointer<ctkSettingsDialog> SettingsDialog;
```

Comment:
Describe the ownership of the settings dialog: when a main window exists and the dialog is displayed then the main window must be set as parent to ensure correct z order; but that also transfers the ownership of the object, therefore we use a smart pointer to keep track if the object is deleted already by the main window.

Reply:
fixed